#!/usr/bin/env python
PACKAGE = "aerial_robot_base"


from dynamic_reconfigure.parameter_generator_catkin import *
from aerial_robot_msgs.msg import DynamicReconfigureLevels
from math import pi

gen = ParameterGenerator()

gen.add("xy_pid_control_flag", bool_t, DynamicReconfigureLevels.RECONFIGURE_CONTROL_FLAG, "XY PID Control Flag", False)

#gen.add("pid_ctrl_loop_rate", double_t, 2, "PidCtrlLoopTate", 0, 0, 10)
gen.add("ctrl_loop_rate", double_t, DynamicReconfigureLevels.RECONFIGURE_CONTROL_LOOP_RATE, "ctrl loop rate", 40, 0, 50)
gen.add("pos_p_gain", double_t, DynamicReconfigureLevels.RECONFIGURE_POS_P_GAIN, "pos p gain", 0.01, 0, 0.2)
gen.add("pos_i_gain", double_t, DynamicReconfigureLevels.RECONFIGURE_POS_I_GAIN, "pos i gain", 0.01, 0, 0.2)
gen.add("pos_i_gain_hover", double_t, DynamicReconfigureLevels.RECONFIGURE_POS_I_GAIN_HOVER, "pos i gain hover", 0.01, 0, 0.2)
gen.add("pos_d_gain", double_t, DynamicReconfigureLevels.RECONFIGURE_POS_D_GAIN, "pos d gain", 0.15, 0, 0.5)
gen.add("vel_p_gain", double_t, DynamicReconfigureLevels.RECONFIGURE_VEL_P_GAIN, "vel p gain", 0.15, 0, 0.5)
gen.add("vel_i_gain", double_t, DynamicReconfigureLevels.RECONFIGURE_VEL_I_GAIN, "vel i gain", 0.001, 0, 0.2)
gen.add("offset", int_t, DynamicReconfigureLevels.RECONFIGURE_OFFSET, "offset", 0, 0, 100)
gen.add("pos_limit", int_t, DynamicReconfigureLevels.RECONFIGURE_POS_LIMIT, "pos limit", 70, 0, 100)
gen.add("pos_p_limit", int_t, DynamicReconfigureLevels.RECONFIGURE_POS_P_LIMIT, "pos p limit", 33, 0, 50)
gen.add("pos_i_limit", int_t,DynamicReconfigureLevels. RECONFIGURE_POS_I_LIMIT, "pos i limit", 33, 0, 50)
gen.add("pos_d_limit", int_t, DynamicReconfigureLevels.RECONFIGURE_POS_D_LIMIT, "pos d limit", 33, 0, 50)
gen.add("i_enable_limit", double_t, DynamicReconfigureLevels.RECONFIGURE_I_ENABLE_LIMIT_HOVER, "i enable limit", 0.4, 0, 2)

exit(gen.generate(PACKAGE, "aerial_robot_base", "XYPidControl"))


